packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -346,7 +347,14 @@ export class AppManager implements BackstageApp {
       const [identityApi, setIdentityApi] = useState<IdentityApi>();
 
       if (!identityApi) {
-        return <Component onSignInSuccess={setIdentityApi} />;
+        // Encapsulate the sign in page component in an API provider which
+        // contains all APIs except the identity API. Provides fast feedback in
+        // case the identity API is accidentally required on the sign-in page.
+        return (
+          <IsolatedApiProvider apis={this.getIdentitylessApiHolder()}>
```

Comment:
Perhaps it would be nicer to achieve this using an IdentityApi implementation that throws a helpful error message instead? It can be added in using a regular ApiProvider then too

Reply:
Mmm, so you mean keep most of the implementation as is (with a separate, isolated API provider wrapping the sign in page), but instead of allowing the default "missing apiRef" error to bubble up, throw in a dummy implementation of IdentityApi that throws a helpful error?

Reply:
Yep, exactly. It can be done with the regular `ApiProvider` though

```tsx
<ApiProvider apis={ApiRegistry.with(identityApiRef, throwingIdentityApi)}>
```

^ overrides the `IdentityApi` implementation and relies on the parent provider for the rest

Reply:
Hmm...  I think this ends up missing the case when an API relies on the identityApi.  Seems like direct `useApi(identityApi)` calls get passed an instance of whatever is provided via this `ApiRegistry.with()` call.  ...But other APIs are pre-instantiated with (and refer to) the `AppIdentityProxy` instance.

Reply:
... typed out a long message here that I forgot to post and lost ._.

Yeah only applying the check to `useApi` is a lil bit silly, because why would you ask for the identity API on the sign-in page. Either way it's a small precaution that might help someone.

Doing the override at the factory level I think is very risky, because you either end up with duplicate implementations, or you may end up with incorrect dependencies, depending on how we'd implement the override. I think a stalled sign-in page is easier to debug and fix compared to for example duplicate implementations, so would rather just leave it at that tbh.

Reply:
> _... typed out a long message here that I forgot to post and lost ._.

I...  Literally just did the exact same thing.  Based on the risks/feedback above: I'm just gonna get rid of all of the app manager changes and will let things hang in the identity proxy.

Reply:
Alright.  Reverted in https://github.com/backstage/backstage/pull/8945/commits/6c59abc74e2148c9e794e990cfd298528fbbc821